bundles/ui/org.eclipse.smarthome.ui.icon/src/main/java/org/eclipse/smarthome/ui/icon/AbstractResourceIconProvider.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.ui.icon;
+
+import java.io.InputStream;
+import java.util.Set;
+
+import org.eclipse.smarthome.ui.icon.IconSet.Format;
+import org.osgi.framework.BundleContext;
+
+/**
+ * This is an abstract base class for implementing icon providers that serve icons from file resources.
+ * These files could be read from the file system, directly from the bundle itself or from somewhere else that can
+ * provide an {@link InputStream}.
+ *
+ * The resources are expected to follow the naming convention "<category>[-<state>].<format>", e.g. "alarm.png" or
+ * "alarm-on.svg".
+ * Resource names must be all lower case. Whether an icon is provided or not is determined by the existence of a
+ * resource without a state postfix.
+ * If a specific resource for a state is available, it will be used. If not, the default icon without a state postfix is
+ * used. If the state is a decimal number between 0 and 100, the implementation will look for a resource with the next
+ * smaller state postfix available. Example: For category "DimmableLight" and state 84, it will check for the resources
+ * dimmablelight-82.png, dimmablelight-81.png, dimmablelight-80.png and return the first one it can find.
```

Comment:
This sounds like it performs really bad. Does it tries to look up all the files? Will check implementation.